shared/pages/Swap/DepositWindow/DepositWindow.js

```diff
@@ -100,25 +100,20 @@ export default class DepositWindow extends Component {
 
     const remainingBalance = new BigNumber(sellAmount).minus(balance)
 
-    if (coinsWithDynamicFee.includes(swap.sellCurrency.toLowerCase())) {
-
-      this.setState(() => ({
-        remainingBalance,
-        dynamicFee,
-      }))
-    } else {
-      this.setState(() => ({
-        remainingBalance,
-      }))
-    }
+    this.setState(() => ({
+      remainingBalance,
+      dynamicFee,
+    }))
   }
 
   getRequiredAmount = async () => {
-    const { swap, sellAmount } =  this.props
+    const { swap } =  this.props
+    const { sellAmount } = this.state
 
     if (coinsWithDynamicFee.includes(swap.sellCurrency.toLowerCase())) {
       const dynamicFee = await helpers[swap.sellCurrency.toLowerCase()].estimateFeeValue({ method: 'swap' })
-      const newSellAmount = sellAmount.plus(dynamicFee)
+
+      const newSellAmount = new BigNumber(sellAmount).plus(dynamicFee)
```

Comment:
```suggestion
      const newSellAmount = BigNumber(sellAmount).plus(dynamicFee)
```
на будущее: у него необязательно вызывать new, возможно, так даже хуже по производительности.